apps/spruce/src/pages/configurePatch/configurePatchCore/useConfigurePatch/index.ts

```diff
@@ -184,59 +71,34 @@ const useConfigurePatch = (patch: ConfigurePatchQuery["patch"]): HookResult => {
       type: "setSelectedAliases",
       aliases,
     });
-  const setSelectedTab = (i: number) =>
-    dispatch({ type: "setSelectedTab", tabIndex: i });
+  const setSelectedTab = (t: ConfigurePatchPageTabs) => {
+    dispatch({ type: "setSelectedTab", tab: t });
+  };
   const setPatchParams = (params: ParameterInput[]) =>
     dispatch({ type: "setPatchParams", params });
 
   useTabShortcut({
-    currentTab: selectedTab,
+    currentTab: tabToIndexMap[state.selectedTab],
     numTabs: indexToTabMap.length,
-    setSelectedTab,
+    setSelectedTab: (i: number) => setSelectedTab(indexToTabMap[i]),
   });
 
+  // Handle redirecting to the correct tab if the tab is not active
+  useEffect(() => {
+    if (!urlTab || !tabToIndexMap[urlTab]) {
+      setSelectedTab(ConfigurePatchPageTabs.Tasks);
```

Comment:
could we group these hooks all together just for organizational purposes? (i.e. put the hooks first, then put the `set` functions after)